service/src/test/java/org/kiwiproject/champagne/dao/BuildDaoTest.java

```diff
@@ -97,17 +99,14 @@ void shouldInsertBuildSuccessfully() {
     class FindPagedBuilds {
 
         @ParameterizedTest
-        @CsvSource(nullValues = "null", value = {
-            "null, null",
-            "null, 42.0",
-            "champagne-service, null",
-            "champagne-service, 42.0"
-        })
-        void shouldReturnListOfBuilds(String componentIdentifierFilter, String componentVersionFilter) {
+        @NullSource
+        @BlankStringSource
+        @ValueSource(strings = { "42.0", "champagne-service" })
+        void shouldReturnListOfBuilds(String componentFilter) {
```

Comment:
Wait...can you do this??? Stack different sources like this? Did you verify this works???

Reply:
Yes, you can! No idea how I never knew this, but `ArgumentsSource` is _repeatable_